src/main/java/sk/uniba/fmph/dcs/game_board/AllPlayersTakeReward.java

```diff
@@ -0,0 +1,60 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.ActionResult;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AllPlayersTakeReward implements EvaluateCivilisationCardImmediateEffect {
+    private final RewardMenu menu;
+    private boolean used = false;
+    private static final int WOODNUMBER = 1;
+    private static final int CLAYNUMBER = 2;
+    private static final int STONENUMBER = 3;
+    private static final int GOLDNUMBER = 4;
+    private static final int TOOLNUMBER = 5;
+    private static final int FIELDNUMBER = 6;
+    private ArrayList<Integer> diceThrows;  //used for tests, null if throws should be random
+
+    public AllPlayersTakeReward(final RewardMenu menu, final ArrayList<Integer> diceThrows) {
+        this.menu = menu;
+        this.diceThrows = diceThrows;
+    }

```

Comment:
```suggestion
    private ArrayList<Integer> diceThrows;
    private final Throw t;

    public AllPlayersTakeReward(final RewardMenu menu, final Throw t) {
        this.menu = menu;
        diceThrows = new ArrayList<>();
        this.t = t;
    }

```
Throw v konstruktore namiesto diceThrows